sdks/python/apache_beam/io/iobase.py

```diff
@@ -1185,6 +1192,9 @@ def check_done(self):
     """
     raise NotImplementedError
 
+  def current_watermark(self):
```

Comment:
Can we return ```None``` instead of ```timestamp.MIN_TIMESTAMP``` by default?  the watermark is required from dynamic split:  https://github.com/apache/beam/blob/fb8efe3af66ddfe3ce7d4020c43d10d8467f0ab9/sdks/python/apache_beam/runners/common.py#L713.  But I don't find anywhere uses the watermark so far.

Reply:
None would actually work better and would imply use the last known value that was output.

It would make sense to [update the proto spec](https://github.com/apache/beam/blob/42ef04ac9b73f72bff7c9427aa835ce3b6f7839c/model/fn-execution/src/main/proto/beam_fn_api.proto#L195) specifically saying that unreported watermarks will use the last value that was reported. If no value has ever been reported, we will default to the [min timestamp](https://github.com/apache/beam/blob/42ef04ac9b73f72bff7c9427aa835ce3b6f7839c/model/pipeline/src/main/proto/beam_runner_api.proto#L39).

Reply:
Seems like we would like to use ```None``` as default value in RestrictionTracker.current_watermark. When constructing [proto](https://github.com/apache/beam/blob/fb8efe3af66ddfe3ce7d4020c43d10d8467f0ab9/sdks/python/apache_beam/runners/common.py#L713), we put min timestamp if current_watermark is ```None```. If that's the case, I'll have a separate PR to do the change. sounds good?

Reply:
sounds good

Reply:
https://github.com/apache/beam/pull/9023